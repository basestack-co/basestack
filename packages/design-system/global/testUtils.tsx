import { type RenderOptions, render } from "@testing-library/react";
import type React from "react";
// Styles
import { ThemeProvider } from "styled-components";
import theme from "../theme/lightTheme";

interface Props {
  children: React.ReactNode;
}

const WithThemeProvider = ({ children }: Props) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithTheme = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: WithThemeProvider, ...options });

export { renderWithTheme };
