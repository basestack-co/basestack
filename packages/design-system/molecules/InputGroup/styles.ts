import styled from "styled-components";
import { space } from "styled-system";

export const Container = styled.div`
  ${space};
`;

export const Header = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: ${({ theme }) => theme.spacing.s2};
`;
