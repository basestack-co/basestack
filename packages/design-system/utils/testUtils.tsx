import defaultTheme from "@basestack/design-system/theme/lightTheme";
import isPropValid from "@emotion/is-prop-valid";
import { RenderResult, render } from "@testing-library/react";
import React from "react";
import { StyleSheetManager, ThemeProvider } from "styled-components";

const Provider = ({
  children,
  theme = defaultTheme,
}: {
  children: React.ReactNode;
  theme: any;
}) => (
  <ThemeProvider theme={theme}>
    <StyleSheetManager shouldForwardProp={isPropValid}>
      {children}
    </StyleSheetManager>
  </ThemeProvider>
);

export const renderWithTheme = (
  ui: React.ReactElement,
  options?: any,
): RenderResult => render(ui, { wrapper: Provider, ...options });
