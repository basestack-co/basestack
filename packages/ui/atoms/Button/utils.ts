import { Size, Target, Type, Variant } from "@/components/atoms/Button/types";
import { css, DefaultTheme } from "styled-components";
import { darken, rem, transparentize } from "polished";

export const handleButtonSize = (size?: Size) => {
  switch (size) {
    case "large":
      return css`
        height: ${rem("60px")};
        font-size: ${rem("18px")};
        line-height: ${rem("25px")};
        padding: 0 ${rem("40px")};
      `;
    case "medium":
      return css`
        height: ${rem("54px")};
        font-size: ${rem("16px")};
        line-height: ${rem("22px")};
        padding: 0 ${rem("34px")};
      `;
    case "normal":
      return css`
        height: ${rem("48px")};
        font-size: ${rem("16px")};
        line-height: ${rem("22px")};
        padding: 0 ${rem("30px")};
      `;
    default:
    case "small":
      return css`
        height: ${rem("46px")};
        font-size: ${rem("14px")};
        line-height: ${rem("19px")};
        padding: 0 ${rem("24px")};
      `;
    case "xSmall":
      return css`
        height: ${rem("42px")};
        font-size: ${rem("14px")};
        line-height: ${rem("19px")};
        padding: 0 ${rem("20px")};
      `;
  }
};

export const handleButtonsColors = (theme: DefaultTheme) => ({
  primary: {
    outline: {
      color: theme.colors.primary,
      borderColor: theme.colors.primary,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.primary,
    },
  },
  secondary: {
    outline: {
      color: theme.colors.secondary,
      borderColor: theme.colors.secondary,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.secondary,
    },
  },
  tertiary: {
    outline: {
      color: theme.colors.gray3,
      borderColor: theme.colors.gray3,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.primary,
      borderColor: "transparent",
      backgroundColor: theme.colors.gray6,
    },
  },
  neutral: {
    outline: {
      color: theme.colors.gray2,
      borderColor: theme.colors.gray2,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.primary,
      borderColor: "transparent",
      backgroundColor: "transparent",
    },
  },
  success: {
    outline: {
      color: theme.colors.success,
      borderColor: theme.colors.success,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.success,
    },
  },
  warning: {
    outline: {
      color: theme.colors.warning,
      borderColor: theme.colors.warning,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.warning,
    },
  },
  danger: {
    outline: {
      color: theme.colors.error,
      borderColor: theme.colors.error,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.error,
    },
  },
});

export const darkenColors = (
  theme: DefaultTheme,
  amount: number,
  type: Type,
  variant: Variant,
  target: Target
) => {
  return darken(
    amount || 0.1,
    handleButtonsColors(theme)[type || "primary"][variant || "filled"][target]
  );
};

export const lightenColors = (
  theme: DefaultTheme,
  amount: number,
  type: Type,
  variant: Variant,
  target: Target
) => {
  return transparentize(
    amount || 0.6,
    handleButtonsColors(theme)[type || "primary"][variant || "filled"][target]
  );
};
