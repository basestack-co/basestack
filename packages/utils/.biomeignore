# Node dependencies
node_modules

# Next.js build output
.next
out

# Environment files
.env
.env.*
!.env.example

# Logs
*.log
npm-debug.log*
yarn-debug.log*
yarn-error.log*
pnpm-debug.log*

# OS / editor files
.DS_Store
Thumbs.db

# Package manager cache
.pnp
.pnp.js
.pnp.cjs

# Test coverage
coverage

# Build artifacts
dist
build

# Temporary files
.tmp
temp