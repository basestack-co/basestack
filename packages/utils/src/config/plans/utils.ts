export const getAppMode = (mode: string) => {
  return (
    ["development", "production", "local"].includes(mode) ? mode : "production"
  ) as "local" | "development" | "production";
};
