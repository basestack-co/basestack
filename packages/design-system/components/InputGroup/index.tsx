import { memo } from "react";
import { SpaceProps } from "styled-system";
import { useTheme } from "styled-components";
import Input, { InputProps } from "../Input";
import Textarea, { TextareaProps } from "../Textarea";
import Text from "../Text";
import { Container, Header } from "./styles";

export interface InputGroupProps extends SpaceProps {
  /**
   * Input Group title
   */
  title?: string;
  /**
   * Input Group hint
   */
  hint?: string;
  /**
   * Input component props
   */
  inputProps?: InputProps;
  /**
   * Textarea component props
   */
  textareaProps?: TextareaProps;
  /**
   * Render textArea
   */
  textarea?: boolean;
  /**
   * Input group label
   */
  label?: string;
}

const InputGroup = ({
  textarea = false,
  inputProps,
  textareaProps,
  title,
  hint,
  label,
  ...props
}: InputGroupProps) => {
  const theme = useTheme();

  return (
    <Container data-testid="input-group-container" {...props}>
      <Header>
        {!!title && (
          <Text fontWeight={500} data-testid="input-group-title" size="small">
            {title}
          </Text>
        )}
        <Text data-testid="input-group-label" muted size="small">
          {label}
        </Text>
      </Header>
      {textarea ? (
        <Textarea {...(textareaProps as TextareaProps)} />
      ) : (
        <Input {...(inputProps as InputProps)} />
      )}
      {!!hint && (
        <Text
          data-testid="input-group-hint"
          size="xSmall"
          mt={theme.spacing.s2}
          color={
            inputProps?.hasError
              ? theme.inputGroup.hint.error.color
              : theme.inputGroup.hint.color
          }
        >
          {hint}
        </Text>
      )}
    </Container>
  );
};

export default memo(InputGroup);
