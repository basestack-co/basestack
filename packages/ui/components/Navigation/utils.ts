// Types
import { Translate } from "next-translate";
import { NextRouter } from "next/router";
// Utils
import { config } from "@basestack/utils";
import { signOut } from "next-auth/react";

export const getInternalLinks = (t: Translate, formId: string) => {
  return [
    {
      icon: "flag",
      activeText: "Submissions",
      to: `/form/${formId}/submissions`,
      text: t("internal.submissions"),
    },
    {
      icon: "flag",
      activeText: "Setup",
      to: `/form/${formId}/setup`,
      text: t("internal.setup"),
    },
    {
      icon: "settings",
      activeText: "Settings",
      to: `/form/${formId}/settings/general`,
      text: t("internal.settings"),
    },
  ];
};

export const getExternalLinks = (t: Translate) => {
  return [
    {
      icon: "description",
      activeText: "Documentation",
      to: config.urls.docs.flags.base,
      isExternal: true,
      text: t("external.docs"),
    },
    {
      icon: "link",
      activeText: "Github",
      to: config.urls.repo,
      isExternal: true,
      text: t("external.github"),
    },
  ];
};

export const getAvatarDropdownList = (
  t: Translate,
  router: NextRouter,
  onCreateForm: () => void,
) => {
  return [
    {
      icon: "add_circle",
      text: t("create.form"),
      onClick: onCreateForm,
      separator: true,
    },
    {
      icon: "settings",
      text: t("dropdown.settings"),
      onClick: () =>
        router.push({
          pathname: "/user/profile/settings",
        }),
    },
    {
      icon: "logout",
      text: t("dropdown.logout"),
      onClick: signOut,
    },
  ];
};
