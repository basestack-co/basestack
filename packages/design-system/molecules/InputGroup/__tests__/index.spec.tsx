import { cleanup } from "@testing-library/react";
import { renderWithTheme } from "../../../utils/testUtils";
import InputGroup from "..";

describe("InputGroup Molecule tests", () => {
  afterEach(cleanup);

  test("should render InputGroup correctly", () => {
    const { asFragment } = renderWithTheme(
      <InputGroup
        title="Feature Key"
        inputProps={{
          onChange: jest.fn(),
          placeholder: "E.g. header_size",
          name: "feature",
          value: "",
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render InputGroup with hint", () => {
    const { asFragment } = renderWithTheme(
      <InputGroup
        title="Feature Key"
        inputProps={{
          onChange: jest.fn(),
          placeholder: "E.g. header_size",
          name: "feature",
          value: "",
        }}
        hint="no special characters"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render InputGroup with label", () => {
    const { asFragment } = renderWithTheme(
      <InputGroup
        title="Feature Key"
        inputProps={{
          onChange: jest.fn(),
          placeholder: "E.g. header_size",
          name: "feature",
          value: "",
        }}
        label="0 / 120"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render InputGroup as textarea", () => {
    const { asFragment } = renderWithTheme(
      <InputGroup
        title="Feature Key"
        textarea
        textareaProps={{
          value: "",
          name: "",
          onChange: jest.fn(),
          placeholder: "Description",
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
