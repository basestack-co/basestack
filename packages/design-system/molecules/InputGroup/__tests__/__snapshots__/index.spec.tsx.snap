// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup Molecule tests should render InputGroup as textarea 1`] = `
<DocumentFragment>
  .c1 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-weight: 500;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c2 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #757575;
}

.c3 {
  border: none;
  background-color: #F6F6F6;
  min-height: 9.375rem;
  max-height: 31.25rem;
  min-width: 100%;
  max-width: 100%;
  border-radius: 0.25rem;
  font-size: 0.875rem;
  font-weight: 400;
  width: 100%;
  padding: 1rem;
  color: #757575;
}

.c3:focus {
  outline: 2px solid #000000;
  color: #000000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0.5rem;
}

<div
    class=""
    data-testid="input-group-container"
  >
    <div
      class="c0"
    >
      <p
        class="c1"
        data-testid="input-group-title"
        font-weight="500"
      >
        Feature Key
      </p>
      <p
        class="c2"
        data-testid="input-group-label"
      />
    </div>
    <textarea
      class="c3"
      data-testid="textarea"
      placeholder="Description"
    />
  </div>
</DocumentFragment>
`;

exports[`InputGroup Molecule tests should render InputGroup correctly 1`] = `
<DocumentFragment>
  .c1 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-weight: 500;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c2 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #757575;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  border: none;
  background-color: #F6F6F6;
  height: 2.75rem;
  border-radius: 0.25rem;
  font-size: 0.875rem;
  font-weight: 400;
  width: 100%;
  padding: 0 1rem;
  color: #757575;
}

.c4:focus {
  outline: 2px solid #000000;
  color: #000000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0.5rem;
}

<div
    class=""
    data-testid="input-group-container"
  >
    <div
      class="c0"
    >
      <p
        class="c1"
        data-testid="input-group-title"
        font-weight="500"
      >
        Feature Key
      </p>
      <p
        class="c2"
        data-testid="input-group-label"
      />
    </div>
    <div
      class="c3"
      data-testid="input-container"
    >
      <input
        class="c4"
        data-testid="input"
        format="normal"
        placeholder="E.g. header_size"
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`InputGroup Molecule tests should render InputGroup with hint 1`] = `
<DocumentFragment>
  .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.75rem;
  line-height: 0.875rem;
  font-weight: 400;
  margin-top: 0.5rem;
  font-family: 'Roboto',sans-serif;
  color: #757575;
}

.c1 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-weight: 500;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c2 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #757575;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  border: none;
  background-color: #F6F6F6;
  height: 2.75rem;
  border-radius: 0.25rem;
  font-size: 0.875rem;
  font-weight: 400;
  width: 100%;
  padding: 0 1rem;
  color: #757575;
}

.c4:focus {
  outline: 2px solid #000000;
  color: #000000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0.5rem;
}

<div
    class=""
    data-testid="input-group-container"
  >
    <div
      class="c0"
    >
      <p
        class="c1"
        data-testid="input-group-title"
        font-weight="500"
      >
        Feature Key
      </p>
      <p
        class="c2"
        data-testid="input-group-label"
      />
    </div>
    <div
      class="c3"
      data-testid="input-container"
    >
      <input
        class="c4"
        data-testid="input"
        format="normal"
        placeholder="E.g. header_size"
        type="text"
      />
    </div>
    <span
      class="c5"
      data-testid="input-group-hint"
    >
      no special characters
    </span>
  </div>
</DocumentFragment>
`;

exports[`InputGroup Molecule tests should render InputGroup with label 1`] = `
<DocumentFragment>
  .c1 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-weight: 500;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c2 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #757575;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  border: none;
  background-color: #F6F6F6;
  height: 2.75rem;
  border-radius: 0.25rem;
  font-size: 0.875rem;
  font-weight: 400;
  width: 100%;
  padding: 0 1rem;
  color: #757575;
}

.c4:focus {
  outline: 2px solid #000000;
  color: #000000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0.5rem;
}

<div
    class=""
    data-testid="input-group-container"
  >
    <div
      class="c0"
    >
      <p
        class="c1"
        data-testid="input-group-title"
        font-weight="500"
      >
        Feature Key
      </p>
      <p
        class="c2"
        data-testid="input-group-label"
      >
        0 / 120
      </p>
    </div>
    <div
      class="c3"
      data-testid="input-container"
    >
      <input
        class="c4"
        data-testid="input"
        format="normal"
        placeholder="E.g. header_size"
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
